### PR TITLE
Updates to CI tests to run regression tests

### PR DESCRIPTION
**Pull request by dustinswales**
This contains changes to one of the CI scripts to compare the regression tests to baseline results. There are two additional python utility scripts needed to complete the workflow.

### CODE REVIEW

File: .github/workflows/ci_run_scm_rts.yml [hunk 1]
```diff
@@ -172,5 +172,30 @@ jobs:
...
         cd ${SCM_ROOT}/scm/bin
-        mkdir ${RUN_DIR} && mkdir ${BIN_DIR}
-        ./run_scm.py --file /home/runner/work/ccpp-scm/ccpp-scm/test/rt_test_cases.py --runtime_mult 0.1
\ No newline at end of file
+        ./run_scm.py --file /home/runner/work/ccpp-scm/ccpp-scm/test/rt_test_cases.py --runtime_mult 0.1
+
+    - name: Gather SCM RT output
+      run: |
+        cd ${SCM_ROOT}/test
+        mkdir /home/runner/work/ccpp-scm/ccpp-scm/test/artifact-${{matrix.build-type}}
```
Comment (grantfirl) [inline .github/workflows/ci_run_scm_rts.yml:180]: Isn't this path defined above as `dir_rt`?

File: .github/workflows/ci_run_scm_rts.yml [hunk 2]
```diff
@@ -172,5 +172,30 @@ jobs:
...
+        ./ci_util.py -b ${{matrix.build-type}}
+
+    - name: Create directory for SCM RT baselines
+      run: mkdir /home/runner/work/ccpp-scm/ccpp-scm/test/BL-${{matrix.build-type}}
```
Comment (grantfirl) [inline .github/workflows/ci_run_scm_rts.yml:184]: Ditto, but with `dir_bl`?

File: .github/workflows/ci_run_scm_rts.yml [hunk 3]
```diff
@@ -172,5 +172,30 @@ jobs:
...
+
+    - name: Download SCM RT baselines
+      run: |
+        cd /home/runner/work/ccpp-scm/ccpp-scm/test/BL-${{matrix.build-type}}
```
Comment (grantfirl) [inline .github/workflows/ci_run_scm_rts.yml:188]: Ditto, dir_bl

File: .github/workflows/ci_run_scm_rts.yml [hunk 4]
```diff
@@ -172,5 +172,30 @@ jobs:
...
+        wget ftp://ftp.rap.ucar.edu:/incoming/irap/met_help/ccpp-scm/rt-baselines-${{matrix.build-type}}.zip
```
Comment (grantfirl) [inline .github/workflows/ci_run_scm_rts.yml:189]: Not sure that this is the best place on the server (`met_help`), but probably doesn't matter too much

File: test/ci_util.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    (build_type) = parse_args()
+
+    #
+    for run in run_list:
+        file_out = "../scm/run/" + "output_"+run["case"]+"_"+run["suite"]+"/output.nc"
+        if exists(file_out):
+            os.system("mkdir -p artifact-"+build_type+"/"+run["case"]+"_"+run["suite"]+"/")
+            os.system("cp " + file_out + " artifact-"+build_type+"/"+run["case"]+"_"+run["suite"]+"/output.nc")
+        else:
+            print("FAIL: Could not copy output to for baseline generation")
```
Comment (grantfirl) [inline test/ci_util.py:35]: one to many prepositions (to for)

File: test/ci_util.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    build_type = args.build_type
+    return (build_type)
+
+def main():
+    
+    (build_type) = parse_args()
+
+    #
+    for run in run_list:
+        file_out = "../scm/run/" + "output_"+run["case"]+"_"+run["suite"]+"/output.nc"
```
Comment (grantfirl) [inline test/ci_util.py:30]: Looks like there is room for defining some common strings to use to clean this up a bit (`run["case"]+"_"+run["suite]`...)

**Review by grantfirl — CHANGES_REQUESTED** [review]
Looks great. I only have minor comments. For posterity, it might be worth adding comments somewhere in the code or PR description about storing baselines as GitHub artifacts, since I doubt many know what that is (me included).

File: .github/workflows/ci_run_scm_rts.yml (hunk 4 above)
Comment (dustinswales) [reply .github/workflows/ci_run_scm_rts.yml:189]: I've been waiting on RAL IT to create a directory for me on the FTP site...

File: .github/workflows/ci_run_scm_rts.yml [hunk 7]
```diff
@@ -186,7 +186,7 @@ jobs:
     - name: Download SCM RT baselines
       run: |
         cd ${dir_bl}
-        wget ftp://ftp.rap.ucar.edu:/incoming/irap/met_help/ccpp-scm/rt-baselines-${{matrix.build-type}}.zip
+        wget ftp://ftp.rap.ucar.edu:/pub/ccpp-scm/rt-baselines-${{matrix.build-type}}.zip
```
Comment (dustinswales) [inline .github/workflows/ci_run_scm_rts.yml:189]: @grantfirl

**Comment by grantfirl** [comment]
@dustinswales We can merge this whenever you're happy with it.

**Pull request merged**